controllers/datasciencecluster/kubebuilder_rbac.go

```diff
@@ -4,6 +4,14 @@ package datasciencecluster
 //+kubebuilder:rbac:groups="datasciencecluster.opendatahub.io",resources=datascienceclusters/finalizers,verbs=update;patch
 //+kubebuilder:rbac:groups="datasciencecluster.opendatahub.io",resources=datascienceclusters,verbs=get;list;watch;create;update;patch;delete
 
+/* Service Mesh prerequisite */
+// +kubebuilder:rbac:groups="maistra.io",resources=servicemeshcontrolplanes,verbs=create;get;list;patch;update;use;watch
```

Comment:
Those are already there. Around line 235.
I think you re-organized in #605. I can reorganize here if you prefer it, or leave as is so that you don't need to fix conflicts on you PR.

Reply:
Leave it, I am anyway not happy with this split anymore. It solved the problem of making controller files easier to read, but not the actual one of having one big dump of all RBAC settings in one place making it hard to navigate. My thinking now is to move relevant permission settings to interested parties e.g. KServe component will have it's own and service mesh setup another one. There's potential to duplicate, but it will make it self-contained and I think controller-gen can sort it out. WDYT? @zdtsw @VaishnaviHire

Reply:
i was trying to have the ones for DSCI only in dsci controller , same concept for DSC and secret-controller, but failed to do this......just way too much need to split. 
i am not against the idea at all to put in the components and controller level

Reply:
+1 for moving them to respective components. We would still need to test if controller-gen updates the role as expected.

I will create a separate GitHub issue for this though